### PR TITLE
Fix VPhys KV

### LINKED ISSUE #472: PhysSceneNode needs to handle binary blobs for vertices
### Describe the bug

Valve updated vphys to store vertices/edges/nodes/etc as binary blobs.

@lewa-j could you take a look?

### What Valve game does this happen in

Dota 2

### Reference file in a Source 2 game

maps/dota.vpk any vphys file in it

### Expected behavior

-

### What version of VRF are you using? On what platform?

ec760fbbc4d254963f0cc17246f4214995db9ed2

### PR DESCRIPTION
**Pull request by moofMonkey**
Fixes #472.

I'm not sure about edges implementation, so it needs to be clarified.

Dota 2 `maps/dota.vpk` `maps/dota/world.vwrld` already opens with current changes.

### CODE REVIEW

File: GUI/Types/Renderer/PhysSceneNode.cs [hunk 1]
```diff
@@ -93,11 +93,26 @@ public PhysSceneNode(Scene scene, PhysAggregateData phys)
...
+                    {
+                        //KV3 has vertices as blob
+                        var verticesBlob = hull.GetArray<byte>("m_Vertices");
+                        vertices = Enumerable.Range(0, verticesBlob.Length / 12)
+                            .Select(i => new Vector3(BitConverter.ToSingle(verticesBlob, i * 12),
+                                BitConverter.ToSingle(verticesBlob, (i * 12) + 4),
+                                BitConverter.ToSingle(verticesBlob, (i * 12) + 8)));
+                    }
                     var vertOffset = verts.Count / 7;
                     foreach (var v in vertices)
```
Comment (xPaw) [inline GUI/Types/Renderer/PhysSceneNode.cs:113]:
```suggestion
                    foreach (var vec in vertices)
```

unnecessary var assignment

Comment (moofMonkey) [reply GUI/Types/Renderer/PhysSceneNode.cs:113]: Nope. I tried doing that.
> CS1656 vec is a foreach iteration variable

**Comment by xPaw** [comment]
maps/dota/phys_level_water.vphys_c is messed up, so edges is unlikely to be correct.

P.S. Wanna come to our discord?

**Comment by xPaw** [comment]
This should be it:
```
                        // 0 - m_nNext, 1 - m_nTwin, 2 - m_nOrigin, 3 - m_nFace
                        var edgesBlob = hull.GetArray<byte>("m_Edges");
                        edges = Enumerable.Range(0, edgesBlob.Length / 4)
                            .Select(i => ((int)edgesBlob[i * 4 + 2], (int)edgesBlob[i * 4 + 1]))
                            .ToArray();
```

but the NTRO test is wrong, it should check whether it's actually a binary blob in kv3.

**Comment by moofMonkey** [comment]
> P.S. Wanna come to our discord?

Joined and verified.

> This should be it:

Yep, that works. Wanna commit it?

> but the NTRO test is wrong, it should check whether it's actually a binary blob in kv3.

Should it be done with try/catch or is there a better way?

**Pull request closed without merging**